app/src/main/java/bacon/john/bitcoinpricewatch/CurrencyDetailFragment.java

```diff
@@ -0,0 +1,143 @@
+package bacon.john.bitcoinpricewatch;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import bacon.john.bitcoinpricewatch.bitcoin.BitcoinCurrencyManager;
+import bacon.john.bitcoinpricewatch.bitcoin.BitcoinCurrencyModel;
+
+/**
+ * A fragment representing a single Currency detail screen.
+ * This fragment is either contained in a {@link CurrencyListActivity}
+ * in two-pane mode (on tablets) or a {@link CurrencyDetailActivity}
+ * on handsets.
+ */
+public class CurrencyDetailFragment extends Fragment {
+    /**
+     * The fragment argument representing the item ID that this fragment
+     * represents.
+     */
+    public static final String CURRENCY_CODE = "currencyCode";
+
+    /**
+     * The content this fragment is presenting.
+     */
+    private BitcoinCurrencyModel mCurrency;
+
+    /**
+     * The timer for fetching the latest info
+     */
+    private Timer mTimer;
+
+    /**
+     * Mandatory empty constructor for the fragment manager to instantiate the
+     * fragment (e.g. upon screen orientation changes).
+     */
+    public CurrencyDetailFragment() {
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (getArguments().containsKey(CURRENCY_CODE)) {
+            // Load the dummy content specified by the fragment
+            // arguments. In a real-world scenario, use a Loader
+            // to load content from a content provider.
+            String currencyCode = getArguments().getString(CURRENCY_CODE);
+            mCurrency = BitcoinCurrencyManager.sharedInstance().getCachedCurrency(currencyCode);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_currency_detail, container, false);
+        return rootView;
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        this.refreshView();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        long delay = 10000;
+        long timeSinceLastUpdate = System.currentTimeMillis() - mCurrency.getUpdatedAt();
+
+        // refresh immediately if it has been more then 10 seconds
+        if (timeSinceLastUpdate >= 10000) {
+            refresh();
+        } else {
+            delay = 10000 - timeSinceLastUpdate;
+        }
+
+        mTimer = new Timer();
+        mTimer.schedule(new TimerTask() {
```

Comment:
since the data on the API updates around every 10 seconds, schedule a timer task that runs that often, and give is a delay that is <= 10 seconds based on has recently we last got an update